docker/docker_image_src.go

```diff
@@ -210,10 +221,11 @@ func (s *dockerImageSource) manifestDigest(ctx context.Context, instanceDigest *
 			return d, nil
 		}
 	}
-	if err := s.ensureManifestIsLoaded(ctx); err != nil {
+	if err := s.ensureManifestIsLoaded(ctx, instanceDigest); err != nil {
```

Comment:
Above this, `manifestDigest` starts with `if instanceDigest != nil { return *instanceDigest }`.  And because `cachedManifest` exists only to avoid fetching the manifest second time below that check, AFAICT we only need the single `cachedManifest`/`cachedManifestMIMEType` value for the primary instance.

So, the only necessary change to this file is AFAICT the `LayerInfosForCopy` API update.

Reply:
Hmm, I'm not sure how to keep GetManifest() from returning the cached value of the primary instance when `instanceDigest` is not nil otherwise.  Should it just be handing it off to `fetchManifest()` directly for those cases?

Reply:
Yes, AFAICS. The comment above `ensureManifestIsLoaded` says that the caching exists only for `GetSignatures(_, nil)` to work reliably.

Manifest caching primarily happens in `image.UnparsedImage`, to benefit all transports.